### ISSUE TITLE
Filtering actions by predicate

### ISSUE THREAD

**Issue by RIAstar** [zalmoxisus/redux-devtools-extension#232]
Currently we seem to have two options for filtering actions - `actionsWhitelist` and `actionsBlacklist` - which can handle globs and regular expressions. 
This allows only for filtering on action types.

It would be very helpful to have a config option like [redux-logger's predicate](https://github.com/evgenyrodionov/redux-logger#log-everything-except-actions-with-certain-type) that allows for more complex filtering based on action payload and/or current state.
For example:
```
predicate: (getState, action) => getState().dev.logLevel === VERBOSE && !action.forwarded
```

As a workaround (without access to state) I tried something like this:
```
{
    actionsFilter : action => predicate(action) ? action : {type: 'BLACKLISTED'},
    actionsBlacklist: ['BLACKLISTED']
}
```
But that didn't work out. It would seem the execution order is working against me, i.e. the blacklist is applied before the transform.

Perhaps you can think of another workaround?

----
Alternatively you could use the existing `actionsFilter` config option and modify it so that it doesn't log anything when the function returns `false`. You'd need to pass the state or a `getState()` function as the third argument to remain backwards compatible. (Personally I'm not a fan of mixed return types though)

**Comment by zalmoxisus** [comment]
@RIAstar, thanks for investigating this.

`actionsFilter` / `statesFilter` are wrong chosen names, which should be renamed to `actionSanitizer` / `stateSanitizer` as in [`redux-remotedev`](https://github.com/zalmoxisus/redux-remotedev#actionsanitizer).

A `predicate` option would be better indeed.

**Comment by RIAstar** [comment]
I imagine that, because this is a browser plugin which can be automatically updated, backwards compatibility is very important.
At best, I suppose you can create an alias and deprecate the old name.

**Comment by zalmoxisus** [comment]
> At best, I suppose you can create an alias and deprecate the old name.

Yes, that's the plan.

**Comment by zalmoxisus** [comment]
@RIAstar, I've implemented the `predicate` parameter and will publish a new version to Chrome Store tomorrow. Meanwhile, you can amend the changes (especially docs).

**Comment by RIAstar** [comment]
Cool! You're fast, man!

**Comment by zalmoxisus** [comment]
Implemented in `2.8.3`.

**Comment by RIAstar** [comment]
Works like a charm. 
Thank you.
